### PR TITLE
Signup: refactor search results logic

### PR DESCRIPTION
**Pull request by ramonjd**
⚠️ This PR is a little on the large side, so if it works okay I can split it up :)

## Changes proposed in this Pull Request

This is an attempt to incorporate many of the suggestions we received in our spike PR #31378, including, but not limited to:

### ~~Debouncing the site preview update so that we don't get 'flashes' of the default site preview as the user types~~

_Update_: We will address this in ~~a future PR~~ https://github.com/Automattic/wp-calypso/pull/32055. See comment: https://github.com/Automattic/wp-calypso/pull/31708#issuecomment-480145319

debounce

### Caching previous results, which we can display to the user if their query yields none (probably should be migrated to the backend eventually)
no-match

### Removing sorting on the clientside
No GIF 🕶 

## Testing instructions
Fire up the branch and search for a vertical. In particular:

* Select a popular category, clear the field, select another
* Enter text, slow and fast
* _More scenarios to come..._

### CODE REVIEW

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 1]
```diff
@@ -83,98 +144,72 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
 		if (
-			hasValue &&
-			valueLengthShouldTriggerSearch &&
-			// Don't trigger a search if there's already an exact, non-user-defined match from the API
-			! result
+			!! value &&
+			value !== this.props.searchValue &&
+			size( value ) >= this.props.charsToTriggerSearch
 		) {
 			this.setState( { railcar: this.getNewRailcar() } );
```
Comment (ramonjd) [inline client/components/site-verticals-suggestion-search/index.jsx]: @adamziel 

Since we're not using this condition block, what would be the conditions for railcar now?

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 2]
```diff
@@ -83,98 +144,72 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
-		const showPopularTopics = shouldShowPopularTopics( this.props.searchValue );
+		const { translate, placeholder, autoFocus, isVerticalSearchPending } = this.props;
 
 		return (
 			<>
-				<QueryVerticals searchTerm={ this.props.searchValue.trim() } debounceTime={ 300 } />
+				<QueryVerticals
+					searchTerm={ this.state.searchValue.trim() }
+					debounceTime={
+						! this.state.isSuggestionSelected || size( this.state.searchValue ) > 4 ? 300 : 0
```
Comment (southp) [inline client/components/site-verticals-suggestion-search/index.jsx]: I personally don't quite understand the purpose behind this condition. Why do we want to alter the debouncing time based on this? How does it reflect on the UX?

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 3]
```diff
@@ -52,9 +50,61 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
+	/***
+	 * Sets `state.results` with incoming HTTP results, retaining previous non-user vertical search results
+	 * if the incoming HTTP results contain only user-defined results.
+	 *
+	 * Then we'll debounce the update on the vertical while the user is typing to reduce
+	 * the amount of times we update the site preview.
+	 *
+	 * This function could better be performed in the backend eventually.
+	 *
+	 * @param {Array} results Incoming HTTP results
```
Comment (southp) [inline client/components/site-verticals-suggestion-search/index.jsx]: One major purpose of the existence of the data layer, redux thunks or any other effectful data request paradigm is to isolate app state and UI components from the data source. In this case, `setSearchResults()` shouldn't be aware of the HTTP thing, but only the data shape of `results`. Thus, let's get rid of all the HTTP-related things in the description :)

File: client/components/site-verticals-suggestion-search/utils.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/** @format */
```
Comment (southp) [inline client/components/site-verticals-suggestion-search/utils.js]: I think this whole module be deprecated now?

**Review by southp — DISMISSED** [review]
The test plan works for me. However, it's quite hard for me to reason the purposes and how each line of change maps to them. If not too much burden, I'd suggest to segregate this PR so that we can have a deeper look into each ingredient and have a better guarantee that nothing unexpected got slipped in.

File: client/components/site-verticals-suggestion-search/README.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ render() {
 
 ## Props
 
-### _(String)_ `initialValue`
+### _(String)_ `searchValue`
 An _optional_ initial value of the search input field. Default is `''`;
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/README.md]: I think `"The search field value."` should be enough for now :)

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 6]
```diff
@@ -63,18 +113,29 @@ export class SiteVerticalsSuggestionSearch extends Component {
 		};
 	}
 
-	// When a user is keying through the results,
-	// only update the vertical when they select a result.
+	/***
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: I'm clearly starstruck 🌟  :)

File: client/components/site-verticals-suggestion-search/utils.js (hunk 4 above)
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/utils.js]: Ah good call!

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 7]
```diff
@@ -52,9 +50,61 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
+			}
+
+			const matches = this.searchForVerticalMatches( this.state.searchValue );
+
+			this.setState( { candidateVerticals: results }, () =>
+				// if a user has selected a vertical from the suggestion list,
+				// set a flag to load sans debounce
+				isSuggestionSelected
+					? this.updateVerticalData( matches, this.state.searchValue )
+					: this.updateVerticalDataDebounced( matches, this.state.searchValue )
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: > why we would need to debounce state updating here

Ooh, that was a bad rebase. You're right @southp . It doesn't need to be there. Nice 👀 !!

But to avoid updating the state too frequently we need it `[onChange](https://github.com/Automattic/wp-calypso/pull/31708/files/d64a4faf7499774bb41d009b7c077a7c61c11526#diff-9553ff7100a8a69d6011c3bc6f8dded2R177)` at least.

**With no debounce:**
no-debounce-onchange

**With:**
debounce-on-change

I'll look more closely at this today.

File: client/components/site-verticals-suggestion-search/index.jsx (hunk 2 above)
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: Me either. I think it was an oversight, that is, I left it in there after test. 

Thanks for catching this (again) :)

File: client/state/signup/verticals/reducer.js [hunk 8]
```diff
@@ -4,14 +4,22 @@
...
-import { SIGNUP_VERTICALS_SET } from 'state/action-types';
+import { SIGNUP_VERTICALS_SET, SIGNUP_VERTICALS_REQUEST } from 'state/action-types';
 
 const verticals = createReducer(
 	{},
 	{
+		[ SIGNUP_VERTICALS_REQUEST ]: state => {
+			return {
+				...state,
+				isFetching: true,
```
Comment (tyxla) [inline client/state/signup/verticals/reducer.js]: I usually advise against any `isFetching`/`isRequesting` when possible. If you're sure we need it and there's no other way - that's fine.

Comment (tyxla) [inline client/state/signup/verticals/reducer.js]: This feels like an unexpected location in the tree for storing a `isFetching` variable. I expected it to be in a separate subtree of the state.

File: client/state/signup/verticals/test/reducer.js [hunk 9]
```diff
@@ -23,6 +38,7 @@ describe( 'state/signup/verticals/reducer', () => {
 			} )
 		).toEqual( {
 			foo: verticals,
+			isFetching: false,
```
Comment (tyxla) [inline client/state/signup/verticals/test/reducer.js]: Is that worth a separate test?

**Review by tyxla — COMMENTED** [review]
This tests great for the Jetpack use case (`/jetpack/connect/site-topic/:site`) 👍 

Left some questions and comments, mostly minor stuff. Nice work!

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 10]
```diff
@@ -83,98 +139,62 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
+	 * @param {Bool} isSuggestionSelected Whether the user has selected a suggestion
+	 */
+	onSiteTopicChange = ( value, isSuggestionSelected ) => {
+		// TODO: Where to put the railcar code will be reconsidered.
 		if (
-			hasValue &&
-			valueLengthShouldTriggerSearch &&
-			// Don't trigger a search if there's already an exact, non-user-defined match from the API
-			! result
+			!! value &&
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: 👍 Good spot.

I hope we can get rid of this condition block all together once the regily question is cleared up.

File: client/components/site-verticals-suggestion-search/README.md [hunk 11]
```diff
@@ -26,8 +26,8 @@ render() {
 
 ## Props
 
-### _(String)_ `initialValue`
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/README.md]: Nothing to see here! :trollface: 

I missed that. Thanks for the 👀

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 12]
```diff
@@ -31,7 +31,6 @@ export class SiteVerticalsSuggestionSearch extends Component {
 		onChange: PropTypes.func,
 		placeholder: PropTypes.string,
 		shouldShowPopularTopics: PropTypes.func,
-		searchValue: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: I've added `searchValue`. `initialValue` was the old prop name. It just doesn't want to die. :)

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 13]
```diff
@@ -52,9 +50,56 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
+		if ( size( results ) ) {
+			const { candidateVerticals, isSuggestionSelected } = this.state;
+			// If the user is typing
+			// and the only result is a user input (non-vertical),
+			// and we have some previous results
+			// then concat that with the previous results and remove the last user input
+			if (
+				! isSuggestionSelected &&
+				! find( results, item => ! item.isUserInputVertical ) &&
+				1 < size( candidateVerticals )
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: Nope. Removing. 👍

File: client/state/signup/verticals/reducer.js (hunk 8 above)
Comment (ramonjd) [reply client/state/signup/verticals/reducer.js]: Totally agree. Will fix this.

Comment (ramonjd) [reply client/state/signup/verticals/reducer.js]: Thanks for raising this. Normally I'd have not added this to the state. There was some headscratching around the problem.

I should be able to wrangle it, but I'll need to add a custom prop method, something like:

```
isVerticalSearchPending: ( searchValue, verticals = [] ) => searchValue && 0 === verticals.length,
```

Or something... :)

**Comment by ramonjd** [comment]
Thanks a lot for the helpful review @tyxla and @southp !!!

I've updated according to your recommendations. 🤞

File: client/components/site-verticals-suggestion-search/index.jsx (hunk 1 above)
Comment (southp) [reply client/components/site-verticals-suggestion-search/index.jsx]: My two cents for `value && value !== this.props.searchValue` since it seems that we can deprecate `chartsToTriggerSearch` . If this sounds good, I think we can also remove that TODO comment. I wrote that because I thought it wasn't the best place, but it now looks about right after your work here :)

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 14]
```diff
@@ -26,35 +26,82 @@ import './style.scss';
...
+		if ( size( results ) ) {
+			const { candidateVerticals, isSuggestionSelected } = this.state;
+			// If the user is typing
+			// and the only result is a user input (non-vertical),
+			// and we have some previous results
+			// then concat that with the previous results and remove the last user input
+			if (
+				! isSuggestionSelected &&
+				! find( results, item => ! item.isUserInputVertical ) &&
+				1 < candidateVerticals.length
```
Comment (southp) [inline client/components/site-verticals-suggestion-search/index.jsx]: FYI no need to use Yoda condition in calypso, but please feel free to keep it if it pleases you :)

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 15]
```diff
@@ -26,35 +26,82 @@ import './style.scss';
...
-		defaultVertical: {},
 	};
 
 	constructor( props ) {
 		super( props );
 		this.state = {
 			railcar: this.getNewRailcar(),
 			candidateVerticals: [],
+			isSuggestionSelected: false,
+			searchValue: props.searchValue,
```
Comment (southp) [inline client/components/site-verticals-suggestion-search/index.jsx]: I'm going to touch one of the hardest CS problem here: Could we name these two state / props differently so we can distinguish them better?

I personally found a few times that I misunderstood the code because I misread `this.state.searchValue` as `this.props.searchValue`, and vice versa. Thus, it would be great if we can have better naming here.

I feel `searchValue` for the state is quite nice, but what would match the meaning of the current `this.props.searchValue` better? In my previous run of refactoring, it becomes a props because the component doesn't hold the input box value itself anymore. Now that the react state one is back with a legit reason (I think it's because debouncing. Please feel free to correct me if I'm wrong),  it's time to revisit the naming and the necessity of this prop.

I don't really know the best direction of this, so this is more of a brain dump comment than a request to fix. Please feel free to keep it this way and we can revisit it in the future.

**Review by southp — COMMENTED** [review]
@ramonjd Thanks for the helpful clarification. I can now understand the whole picture of this PR. With  the in-depth look of @tyxla , there is also definitely no need to split anymore :)

It works pretty well for me, both on the onboarding flow and the main flow. I've noted a few more places that I missed previously and would worth polishing. Please feel free to DM me if you want a timely revisit 🙇🏻

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 16]
```diff
@@ -26,35 +26,82 @@ import './style.scss';
 
 export class SiteVerticalsSuggestionSearch extends Component {
 	static propTypes = {
+		autoFocus: PropTypes.bool,
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/index.jsx]: We don't seen to be documenting `autoFocus` in the README - now is a good chance to do that.

**Review by tyxla — COMMENTED** [review]
I think this is getting closer 👍 

I agree with @southp's feedback, in which are brought some really good points.

Haven't re-tested, but will do on the next round of reviews. Left some more comments in the meantime.

**Comment by ramonjd** [comment]
Great feedback @southp and @tyxla 

Looking at this today. 

Happy that we can address these things now, and not after merge! 

Getting closer... :)

File: client/components/site-verticals-suggestion-search/index.jsx (hunk 14 above)
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: I know, it's a habit that's hard to break now, like ( spaces ) in all my ( parentheses ) 😆

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 17]
```diff
@@ -184,15 +213,16 @@ export default localize(
...
 				verticals: verticals || [],
-				isVerticalSearchPending,
 				defaultVertical: get( getVerticals( state, 'business' ), '0', {} ),
 			};
 		},
 		( dispatch, ownProps ) => ( {
+			bindDebounce: func => debounce( func, 1000 ),
 			shouldShowPopularTopics: searchValue => ! searchValue && ownProps.showPopular,
+			isVerticalSearchPending: ( searchValue, verticals = [] ) =>
+				searchValue && 0 === verticals.length,
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: > If we have to mock them in unit tests, define them through defaultProps could work.

That was the main motivation, particularly for the `bindDebounce`. We could also extract them into a `utils.js` and mock the import methods.

👍

[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: Or just mock `debounce` 🎉

File: client/components/site-verticals-suggestion-search/popular-topics.jsx [hunk 18]
```diff
@@ -65,7 +65,7 @@ class PopularTopics extends PureComponent {
 	onClick = event => {
 		event.preventDefault();
 		event.stopPropagation();
-		this.props.onSelect( event.currentTarget.value );
+		this.props.onSelect( event.currentTarget.value, true );
```
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/popular-topics.jsx]: > If I understand correctly, the reason that we are passing true here is to comply with the parameter list of onChange() of <SuggestionSearch/>

I was trying to conform to the `isNavigating` bool. I can extract that as you suggested and resolve it in the consuming component. 🙇

File: client/components/site-verticals-suggestion-search/index.jsx (hunk 15 above)
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: I think it was a product of the rebase after the state tree updates.

I'll change it back to my original name `initialValue`, which describes the prop more accurately now since we're using the internal state to track updates.

Cheers!

File: client/components/site-verticals-suggestion-search/index.jsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: Hmmm, anything is possible. If it can't find a matching vertical, the API will always return one result, that is, whatever the user has inputted. 

But it won't hurt to be defensive here.

👍

[the remark this answers is not in the document]
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/index.jsx]: Now that I think of it, I think that was one of the reasons why I added `isFetching` to the state 🤣

File: client/components/site-verticals-suggestion-search/README.md [hunk 19]
```diff
@@ -15,9 +15,9 @@ onChange( { vertical_name, vertical_slug, vertical_id } ) {
 render() {
 	return (
 		<SiteVerticalsSuggestionSearch
-			onChange={ this.onChange }
-			initialValue={ this.state.initialValue }
-			charsToTriggerSearch={ 3 }
+			autoFocus={ true }
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/README.md]: Nit: `={ true }` isn't necessary; we could just have `autoFocus` and it would do the same thing.

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 20]
```diff
@@ -26,55 +26,126 @@ import './style.scss';
...
+		id: `${ uuid().replace( /-/g, '' ) }-site-vertical-suggestion`,
+		fetch_algo: '/verticals',
+		action: 'site_vertical_selected',
+	} );
+
+	/**
+	 * Searches the API results for a direct match on the user search query.
+	 *
+	 * @param {String} value Search query array
+	 * @returns {Object?} An object from the vertical results array
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/index.jsx]: Docs nit: unless I'm missing something, this function will return `undefined` if there are no matches, and not `null`. We should reflect that in the docs (removing the `?`).

In case I'm right about this, let's make sure any usages of `searchForVerticalMatches` work well with the `undefined` case.

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 21]
```diff
@@ -83,116 +154,74 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
-			isNavigating,
-		} );
+		// We debounce the update on the vertical while the user is typing
+		// to prevent unnecessary site preview updates
+		this.setState(
+			{
+				isSuggestionSelected,
+				inputValue: value,
+			},
+			() => this.updateVerticalDataDebounced( this.searchForVerticalMatches( value ), value )
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/index.jsx]: Since we'll debounce this anyway, could we move this as a separate call after `this.setState()`? When the call is debounced for a second, there's little difference if we wait or not for the `setState()` to finish its job.

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 22]
```diff
@@ -26,55 +26,126 @@ import './style.scss';
...
-		initialValue: PropTypes.string,
+		autoFocus: PropTypes.bool,
+		defaultVertical: PropTypes.object,
 		onChange: PropTypes.func,
+		onChangeDebounceFn: PropTypes.func,
 		placeholder: PropTypes.string,
-		shouldShowPopularTopics: PropTypes.func,
 		searchValue: PropTypes.string,
+		showPopular: PropTypes.bool,
+		shouldShowPopularTopics: PropTypes.func,
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/index.jsx]: Seems like this one is no longer used in favor of the corresponding method.

File: client/components/site-verticals-suggestion-search/README.md [hunk 23]
```diff
@@ -58,4 +52,9 @@ Returns _{Object}_:
 
 ```
 
+### _(Boolean)_ `showPopular` 
+_Optional_ Informs the component whether to show a list of popular vertical topics when the input field is empty. Default: `false`
+
+### _(Boolean)_ `autoFocus` 
+_Optional_ When set to `true` gives immediate focus to the search input field. Default: `false`
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/README.md]: Would be great if we maintain a thorough list of all props here. Definitely don't consider that as a requirement though - it's something that can be followed up with in another PR.

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 24]
```diff
@@ -26,55 +26,126 @@ import './style.scss';
...
-	updateVerticalData = ( result, value ) =>
+	/**
+	 * Callback to be passed to consuming component when the search value is updated.
+	 * TODO: once the siteVertical state got simplified, this can be removed.
+	 *
+	 * @param {Object} verticalData An object from the vertical results array
+	 * @param {String} value Search query array
+	 */
+	updateVerticalData = ( verticalData, value = '' ) => {
+		value = value.trim();
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/index.jsx]: Oops. Let's avoid mutating the argument 😉 Creating a copy of it in another variable is fine.

File: client/components/site-verticals-suggestion-search/index.jsx [hunk 25]
```diff
@@ -83,116 +154,74 @@ export class SiteVerticalsSuggestionSearch extends Component {
...
-			hasValue &&
-			valueLengthShouldTriggerSearch &&
-			// Don't trigger a search if there's already an exact, non-user-defined match from the API
-			! result
-		) {
+	/**
+	 * Callback to be passed to consuming component when the search field is updated.
+	 *
+	 * @param {String} value The new search value
+	 * @param {Bool} isSuggestionSelected Whether the user has selected a suggestion
```
Comment (tyxla) [inline client/components/site-verticals-suggestion-search/index.jsx]: Nit of the nits: I've found that vertically aligning the different fragments of the JSdoc makes it so much more readable:

```
	 * @param {String} value                The new search value
	 * @param {Bool}   isSuggestionSelected Whether the user has selected a suggestion
```

I'm fine either way tho - feel free to take this suggestion or throw it in the trash.

**Review by tyxla — COMMENTED** [review]
Great work!

I assume this is still not ready for review, but I've left some more little feedback that could be helpful in the meantime. 90% is nits 😉 

Let me know when this is ready for another 👁

File: client/components/site-verticals-suggestion-search/README.md (hunk 23 above)
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/README.md]: I've taken the approach that the README is for the consumer of the component, and therefore lists the props required to embed it, not necessarily those defined via connect, which are only important for internals. 

Having said that, happy to add them later. I'm sure I'll be back in this neck of woods very soon! :)

File: client/components/site-verticals-suggestion-search/README.md (hunk 19 above)
Comment (ramonjd) [reply client/components/site-verticals-suggestion-search/README.md]: Yeah, I'm being intentionally explicit here. 👍

**Comment by ramonjd** [comment]
Thanks for helping make this patch better @tyxla and @southp 🙇 🙇 🙇 🙇 🙇 🙇 🙇

All suggestions implemented gladly :)

@southp  Any chance you might give it a whirl today for a possible launch?

**Comment by ramonjd** [comment]
The latest commit removes the input debounce for now, to be revisited later.

The main reason for the debounce was to prevent the site preview mockup from unsightly 'flashes' due to hammering the state with updates.

We have to remove it however because we're tracking a user's selection `onSubmit`. 

This tracking event requires vertical data related to the user's input. 

Due to the debounce however, the field value is often added to the state too late. We need to make sure we have the latest response that corresponds to the user input before the user can submit the form.

Because of the input field debounce, we're not updating the state tree in time for `null == getVertical( state, searchValue )` to have any effect, that is, until the debounce is flushed, the field and the state are not synced. The result is that we send of stale data:

Apr-05-2019 14-11-35

I think this indicates that we need to address the mockup flashes probably in the mockup itself rather than in `<SiteVerticalsSuggestionSearch />`.

File: client/signup/steps/site-topic/form.jsx [hunk 26]
```diff
@@ -101,8 +101,7 @@ class SiteTopicForm extends Component {
 export default connect(
 	state => {
 		const siteTopic = getSiteVerticalName( state );
-		const isVerticalSearchPending = null == getVerticals( state, siteTopic );
-		const isButtonDisabled = ! siteTopic || isVerticalSearchPending;
+		const isButtonDisabled = ! siteTopic || null == getVerticals( state, siteTopic );
```
Comment (tyxla) [inline client/signup/steps/site-topic/form.jsx]: I assume the `null ==` here is intentionally not strict?

**Review by tyxla — APPROVED** [review]
Nice work 👍 Code looking good and working well for the Jetpack step.
